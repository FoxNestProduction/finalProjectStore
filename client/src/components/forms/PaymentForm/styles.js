export const title = {
  color: 'text.primary',
  fontSize: {
    mobile: '32px',
    tablet: '40px',
    desktop: '48px',
  },
  maxWidth: {
    lgTablet: '500px',
    desktop: '570px',
  },
  mr: {
    lgTablet: '210px',
    desktop: '270px',
  },
};

export const starsWrapper = {
  display: {
    mobile: 'none',
    lgTablet: 'block',
  },
  position: 'absolute',
  top: '0',
  right: '0',
};

export const inputsWrapper = {
  maxWidth: {
    lgTablet: '500px',
    desktop: '570px',
  },
  m: {
    tablet: '0 auto',
    lgTablet: '0 210px 0 0',
    desktop: '0 270px 0 0',
  },
};

export const subtitle = {
  mb: '30px',
  color: 'text.primary',
  fontWeight: 'fontWeightMedium',
  fontSize: {
    mobile: '20px',
    tablet: '24px',
    desktop: '26px',
  },
};

export const paymentWrapper = {
  '&.MuiFormControl-root': {
    mt: '18px',
  },
};

export const paymentRadioBtn = {
  '& .MuiTypography-root': {
    fontSize: {
      mobile: 14,
      tablet: 16,
      desktop: 18,
    },
  },
  '& .MuiSvgIcon-root': {
    fontSize: {
      tablet: 25,
    },
  },
};

export const buttonsWrapper = {
  mt: {
    mobile: '55px',
    lgTablet: '65px',
  },
  display: 'flex',
  justifyContent: 'space-between',
};

export const btn = {
  p: '12px',
  borderRadius: '16px',
  width: {
    mobile: '140px',
    tablet: '150px',
    desktop: '170px',
  },
  fontSize: {
    tablet: '18px',
    desktop: '24px',
  },
};

export const continueBtn = {
  color: 'text.primaryLight',
  transition: 'background-color 0.3s ease, box-shadow 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
  },
};

export const backBtn = {
  mr: '20px',
  color: 'text.primary',
  border: 1,
  transition: 'background-color 0.3s ease, color 0.3s ease',
  ':hover': {
    backgroundColor: 'primary.hover',
    color: 'text.primaryLight',
  },
};
