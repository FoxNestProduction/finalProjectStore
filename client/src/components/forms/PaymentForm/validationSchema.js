import { object, string } from 'yup';

const validationSchema = object({
  name: string()
    .required('Name is required'),
});

export default validationSchema;
