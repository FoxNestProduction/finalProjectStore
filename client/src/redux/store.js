import { configureStore, getDefaultMiddleware, combineReducers } from '@reduxjs/toolkit';
// todo: правила для redux-persist
// eslint-disable-next-line import/no-extraneous-dependencies
import storage from 'redux-persist/lib/storage';
// eslint-disable-next-line import/no-extraneous-dependencies
import { persistReducer } from 'redux-persist';
import modalSlice from './slices/modalSlice';
import productsSlice from './slices/productsSlice';
import authorizationSlice from './slices/authorizationSlice';
import userSlice from './slices/userSlice';
import partnersSlice from './slices/partnersSlice';
import searchSlice from './slices/searchSlice';
import errorSlice from './slices/errorSlice';
import reviewsSlice from './slices/reviewsSlice';
import favouriteSlice from './slices/favouriteSlice';

const authPersistConfig = {
  key: 'authorization',
  version: 1,
  storage,
};

const userPersistConfig = {
  key: 'user',
  version: 1,
  storage,
};

const searchPersistConfig = {
  key: 'search',
  version: 1,
  storage,
};
const favouritePersistConfig = {
  key: 'favorites',
  version: 1,
  storage,
};

/*
    todo: - ЗВЕРНІТЬ УВАГУ!
      якщо вам потрібно щоб ваш стейт не зникав після перезавантаження
      то юзайте підхід як authorization і user,
      якщо ж ні то варіант написання як для modal всередині reducer
      Якщо що пишіть @Ihor_Kacher
*/
const reducer = combineReducers({
  error: errorSlice,
  modal: modalSlice,
  partners: partnersSlice,
  authorization: persistReducer(authPersistConfig, authorizationSlice),
  user: persistReducer(userPersistConfig, userSlice),
  products: productsSlice,
  search: persistReducer(searchPersistConfig, searchSlice),
  reviews: reviewsSlice,
  favourites: persistReducer(favouritePersistConfig, favouriteSlice),
});

const middleware = getDefaultMiddleware({
  serializableCheck: false,
});

const store = configureStore({
  reducer,
  middleware,
});

export default store;
