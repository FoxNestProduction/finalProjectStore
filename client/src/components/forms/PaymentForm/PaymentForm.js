/* eslint-disable max-len */
/* eslint-disable no-undef */
import React from 'react';
import { Form, Formik } from 'formik';
import Stack from '@mui/material/Stack';
import {
  Typography,
} from '@mui/material';
import { useNavigate } from 'react-router';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import Box from '@mui/material/Box';
import IconButton from '@mui/material/IconButton';
import CheckCircleOutlineIcon from '@mui/icons-material/CheckCircleOutline';
import RadioButtonUncheckedIcon from '@mui/icons-material/RadioButtonUnchecked';
import Input from '../../inputs/Input/Input';
import { subtitle, input, paymentSystemsWrapper, imgVisa } from './styles';
import CheckoutActions from '../CheckoutForm/CheckoutActions';
import { setConfirmedOrder } from '../../../redux/slices/orderSlice';
import { removeDataFromSessionStorage } from '../../../utils/sessionStorageHelpers';
import { CHECKOUT_LS_KEY } from '../../../constants';
import saveUserInfoToSessionStorage from '../../../utils/saveUserInfoToSessionStorage';
import { instance } from '../../../API/instance';
import { resetCart } from '../../../redux/slices/cartSlice';

const PaymentForm = () => {
  const navigate = useNavigate();
  const isUserAuthorized = useSelector((state) => state.authorization.isUserAuthorized);
  const user = useSelector((state) => state.user.user, shallowEqual);
  const orderInfo = useSelector((state) => state.order.orderInfo, shallowEqual);
  const dispatch = useDispatch();

  const initialValues = {
    name: isUserAuthorized ? `${user.firstName} ${user.lastName}` : `${orderInfo.name}`,
    cardNumber: '0000 0000 0000 0000',
    expiryDate: '03/25',
    cvv: '123',
  };

  const handleContinue = async (values, actions) => {
    try {
      const response = await instance.post('/orders', orderInfo);
      console.log(response);
      dispatch(setConfirmedOrder(response.data.order));
      removeDataFromSessionStorage(CHECKOUT_LS_KEY);
      dispatch(resetCart());
      if (isUserAuthorized && user) {
        saveUserInfoToSessionStorage(user);
      }
      navigate('/order-confirmation');
    } catch (err) {
      console.log('Error placing new order: ', err);
    }
  };

  return (
    <Formik
      initialValues={initialValues}
      onSubmit={handleContinue}
    >
      {({ isValid }) => (
        <Form>
          <Stack
            spacing={4}
          >
            <Box sx={paymentSystemsWrapper}>
              <IconButton
                aria-label="checked"
                size="small"
                disableFocusRipple
                disableRipple
                sx={{
                  cursor: 'initial',
                }}
                disabled
              >
                <CheckCircleOutlineIcon fontSize="small" />
              </IconButton>
              <Box
                component="img"
                src={`${process.env.PUBLIC_URL}/img/checkout/visa.png`}
                alt="visa"
                sx={imgVisa}
              />
              <Typography
                variant="body1"
                component="p"
                sx={{ userSelect: 'none', color: 'text.primary' }}
              >
                Credit /
                {' '}
                <Typography
                  component="span"
                  sx={{ whiteSpace: 'nowrap' }}
                >
                  Debit card
                </Typography>
              </Typography>
            </Box>
            <Box sx={paymentSystemsWrapper}>
              <IconButton
                aria-label="checked"
                size="small"
                disableFocusRipple
                disableRipple
                sx={{
                  cursor: 'initial',
                }}
                disabled
              >
                <RadioButtonUncheckedIcon fontSize="small" />
              </IconButton>
              <Box
                component="img"
                src={`${process.env.PUBLIC_URL}/img/checkout/mastercard.png`}
                alt="visa"
                sx={imgVisa}
              />
              <Typography
                variant="body1"
                component="p"
                sx={{ userSelect: 'none', color: 'text.primary' }}
              >
                Credit /
                {' '}
                <Typography
                  component="span"
                  sx={{ whiteSpace: 'nowrap' }}
                >
                  Debit card
                </Typography>
              </Typography>
            </Box>

            <Typography variant="h3" component="h2" align="left" sx={subtitle}>
              Add new card
            </Typography>

            <Input
              name="name"
              id="checkout-name"
              label="Cardholder name*"
              bgColor="#FFF"
              disabled
              styles={input}
            />
            <Input
              name="cardNumber"
              id="checkout-cardNumber"
              label="Card number*"
              bgColor="#FFF"
              disabled
              styles={input}
            />
            <Box sx={{ display: 'flex', gap: '5%' }}>
              <Input
                name="expiryDate"
                id="checkout-expiryDate"
                label="Expiry date"
                bgColor="#FFF"
                disabled
                styles={input}
              />
              <Input
                name="cvv"
                id="checkout-cvv"
                label="CVV"
                bgColor="#FFF"
                type="password"
                disabled
                styles={input}
              />
            </Box>

          </Stack>
          <CheckoutActions isValid={isValid} />
        </Form>
      )}
    </Formik>
  );
};

export default PaymentForm;
