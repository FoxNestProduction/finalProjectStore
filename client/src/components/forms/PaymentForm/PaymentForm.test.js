import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import PaymentForm from './PaymentForm';
import store from '../../../redux/store';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router', () => ({
  ...jest.requireActual('react-router'),
  useNavigate: jest.fn(),
}));

describe('Payment form component', () => {
  test('Payment form snapshot', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <PaymentForm />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('displays payment form correctly', () => {
    render(
      <Provider store={store}>
        <PaymentForm />
      </Provider>,
    );

    expect(screen.getByText('Add new card')).toBeInTheDocument();
    expect(screen.getByLabelText('Cardholder name*')).toBeInTheDocument();
    expect(screen.getByLabelText('Card number*')).toBeInTheDocument();
    expect(screen.getByLabelText('Expiry date')).toBeInTheDocument();
    expect(screen.getByLabelText('CVV')).toBeInTheDocument();
    expect(screen.getByText('Back')).toBeInTheDocument();
    expect(screen.getByText('Continue')).toBeInTheDocument();
  });
});
