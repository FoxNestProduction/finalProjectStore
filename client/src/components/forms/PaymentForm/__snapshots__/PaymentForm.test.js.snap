// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment form component Payment form snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiStack-root css-8mcjw0-MuiStack-root"
    >
      <div
        class="MuiBox-root css-1e53dx2"
      >
        <button
          aria-label="checked"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-c9qrlw-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="CheckCircleOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </button>
        <img
          alt="visa"
          class="MuiBox-root css-1j3jcp4"
          src="/img/checkout/visa.png"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1271keo-MuiTypography-root"
        >
          Credit / 
          <span
            class="MuiTypography-root MuiTypography-body1 css-rhzpb0-MuiTypography-root"
          >
            Debit card
          </span>
        </p>
      </div>
      <div
        class="MuiBox-root css-1e53dx2"
      >
        <button
          aria-label="checked"
          class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-c9qrlw-MuiButtonBase-root-MuiIconButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
            data-testid="RadioButtonUncheckedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </button>
        <img
          alt="visa"
          class="MuiBox-root css-1j3jcp4"
          src="/img/checkout/mastercard.png"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-1271keo-MuiTypography-root"
        >
          Credit / 
          <span
            class="MuiTypography-root MuiTypography-body1 css-rhzpb0-MuiTypography-root"
          >
            Debit card
          </span>
        </p>
      </div>
      <h2
        class="MuiTypography-root MuiTypography-h3 MuiTypography-alignLeft css-1nzy6yt-MuiTypography-root"
      >
        Add new card
      </h2>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="checkout-name"
        >
          Cardholder name*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wvatql-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id="checkout-name"
            name="name"
            type="text"
            value="undefined"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Cardholder name*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="checkout-cardNumber"
        >
          Card number*
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wvatql-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-describedby="helper-text"
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled=""
            id="checkout-cardNumber"
            name="cardNumber"
            type="text"
            value="0000 0000 0000 0000"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Card number*
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
          id="helper-text"
        />
      </div>
      <div
        class="MuiBox-root css-l7qti"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="checkout-expiryDate"
          >
            Expiry date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl css-1wvatql-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-describedby="helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id="checkout-expiryDate"
              name="expiryDate"
              type="text"
              value="03/25"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Expiry date
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
            id="helper-text"
          />
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="checkout-cvv"
          >
            CVV
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-s942kh-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-describedby="helper-text"
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              disabled=""
              id="checkout-cvv"
              name="cvv"
              type="password"
              value="123"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <button
                aria-label="toggle password visibility"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1afvcqy-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VisibilityIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  CVV
                </span>
              </legend>
            </fieldset>
          </div>
          <p
            class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained MuiFormHelperText-filled css-1wc848c-MuiFormHelperText-root"
            id="helper-text"
          />
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-qga54k"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1x2zd9l-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Back
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-r27lj0-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Continue
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
